services/viva-ms/helpers/populateForm.js

```diff
@@ -0,0 +1,34 @@
+import { generateDataMap, formatAnswer, mergeAnswers } from '../../../libs/formAnswers';
+
+export function populateChildrenAnswers(repeaterInputList, childrenList) {
+  const answers = [];
+
+  childrenList.forEach((child, childrenListIndex) => {
```

Comment:
This conversion from childrenList to answerList could be done in a more modern fashion by using reduce/map.
What we want is to create answers from children and repeaterInput values. This is a perfect scenario for the usage of the reduce function.

```JS
const children = [{
  role: 'child',
  personalNumber: 1990123456,
  firstName: 'John',
  lastName: 'Doe',
}, ...]
const answers = children.reduce((answers, child, childIndex) => {
  
  const childAnswers = Object(child).keys.map(key => {
     // create child answer from inputs
     const answer = repeaterInputList.map((input) => {
         if (input.tags.includes(key){
           const inputId = input.id.replace('[*]', childIndex)
           return formatAnswer(inputId, input.tags, child[key])
         }
       }
     )  
  })
  
  return [...answers, ...childAnswers]
}, [])

```

Reply:
Refactored. Not sure if it's less readable :)